api-mock/api-mock-server.ts

```diff
@@ -111,68 +285,127 @@ app.use((req, res, next) => {
+    resetVars();
+    res.json({ ok: true });
+});
+
+app.post("/mock/mockexternalapi", async (req, res, next) => {
+    const { url, status, body, path, method } = req.body;
+    nock(url)[method](path).reply(status, body);
+    res.json({ ok: true });
+});
+
+app.get("/mock/waitforevent", async (req, res, next) => {
     try {
-        log("EmailPassword:signup %j", req.body);
-        const response = await EmailPassword.signUp(
-            req.body.tenantId || constrains.tenantId,
-            req.body.email,
-            req.body.password,
-            req.body.session,
-            req.body.userContext
-        );
-        res.json(response);
+        log("ProcessState:waitForEvent %j", req.query);
+        const instance = ProcessState.getInstance();
+        const eventEnum = req.query.event ? Number(req.query.event) : null;
+        const event = eventEnum ? await instance.waitForEvent(eventEnum) : undefined;
+        res.json(event);
     } catch (e) {
-        log(e);
         next(e);
     }
 });
 
-app.post("/mock/EmailPassword/createResetPasswordLink", async (req, res, next) => {
+setupSupertokensRoutes(app, log);
+setupEmailpasswordRoutes(app, log);
+setupAccountlinkingRoutes(app, log);
+setupSessionRoutes(app, log);
+setupEmailverificationRoutes(app, log);
+setupMultitenancyRoutes(app, log);
+setupPasswordlessRoutes(app, log);
+setupMultiFactorAuthRoutes(app, log);
+setupThirdPartyRoutes(app, log);
+setupTOTPRoutes(app, log);
+
+// *** Custom routes to help with session tests ***
+app.post("/create", async (req, res, next) => {
     try {
-        log("EmailPassword:createResetPasswordLink %j", req.body);
-        const response = await EmailPassword.createResetPasswordLink(
-            req.body.tenantId || constrains.tenantId,
-            req.body.userId,
-            req.body.email,
-            req.body.userContext
-        );
-        res.json(response);
-    } catch (e) {
-        log(e);
-        next(e);
+        let recipeUserId = supertokens.convertToRecipeUserId(req.body.recipeUserId);
+        await Session.createNewSession(req, res, "public", recipeUserId, {}, {});
+        res.status(200).send("");
+    } catch (error) {
+        next(error);
     }
 });
-
-app.post("/mock/AccountLinking/createPrimaryUser", async (req, res, next) => {
+app.post("/getsession", async (req, res, next) => {
     try {
-        log("AccountLinking:createPrimaryUser %j", req.body);
-        const recipeUserId = STExpress.convertToRecipeUserId(req.body.recipeUserId);
-        const response = await AccountLinking.createPrimaryUser(recipeUserId, req.body.userContext);
-        res.json(response);
-    } catch (e) {
-        next(e);
+        let session = await Session.getSession(req, res);
+        res.status(200).json({
+            userId: session.getUserId(),
+            recipeUserId: session.getRecipeUserId().getAsString(),
+        });
+    } catch (error) {
+        next(error);
```

Comment:
I think just letting these throw works as well.

Reply:
with async function it doesnt